Shipkit 2.0.12 does not work in mockito project

While trying to make use of shipkit to 2.0.12 in mockito project we headed into
```
    ./gradlew performRelease
[performRelease] Parallel execution is an incubating feature.
[performRelease]   Building version '2.17.2' (value loaded from 'version.properties' file).
[performRelease] 
[performRelease] FAILURE: Build failed with an exception.
[performRelease] 
[performRelease] * What went wrong:
[performRelease] Circular dependency between the following tasks:
[performRelease] :gitPush
[performRelease] +--- :android:bintrayUpload
[performRelease] |    \--- :gitPush (*)
[performRelease] +--- :deprecatedPluginsTest:bintrayUpload
[performRelease] |    \--- :gitPush (*)
[performRelease] +--- :extTest:bintrayUpload
[performRelease] |    \--- :gitPush (*)
[performRelease] +--- :inline:bintrayUpload
[performRelease] |    \--- :gitPush (*)
[performRelease] +--- :junit-jupiter:bintrayUpload
[performRelease] |    \--- :gitPush (*)
[performRelease] +--- :kotlinTest:bintrayUpload
[performRelease] |    \--- :gitPush (*)
[performRelease] \--- :testng:bintrayUpload
[performRelease]      \--- :gitPush (*)
[performRelease] 
[performRelease] (*) - details omitted (listed previously)
```

see https://travis-ci.org/mockito/mockito/jobs/358045381#L1347-L1379 for a full build log.

--- a/subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/bintray/BintrayReleasePlugin.java
+++ b/subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/bintray/BintrayReleasePlugin.java
@@ -14,6 +14,7 @@ import org.shipkit.internal.gradle.release.ReleasePlugin;
 import org.shipkit.internal.gradle.util.BintrayUtil;
 
 import static org.shipkit.internal.gradle.configuration.DeferredConfiguration.deferredConfiguration;
+import static org.shipkit.internal.gradle.java.JavaPublishPlugin.MAVEN_LOCAL_TASK;
 
 /**
  * Configures Java project for automated releases with Bintray.
@@ -37,31 +38,39 @@ public class BintrayReleasePlugin implements Plugin<Project> {
         project.getPlugins().apply(ReleasePlugin.class);
         final ShipkitConfiguration conf = project.getPlugins().apply(ShipkitConfigurationPlugin.class).getConfiguration();
 
-        project.allprojects(subproject -> subproject.getPlugins().withType(ShipkitBintrayPlugin.class, plugin -> {
-            Task bintrayUpload = subproject.getTasks().getByName(ShipkitBintrayPlugin.BINTRAY_UPLOAD_TASK);
-            Task performRelease = project.getTasks().getByName(ReleasePlugin.PERFORM_RELEASE_TASK);
-            performRelease.dependsOn(bintrayUpload);
+        final Task gitPush = project.getTasks().getByName(GitPlugin.GIT_PUSH_TASK);
+        final Task performRelease = project.getTasks().getByName(ReleasePlugin.PERFORM_RELEASE_TASK);
 
-            //Making git push run as late as possible because it is an operation that is hard to reverse.
-            //Git push will be executed after all tasks needed by bintrayUpload
-            // but before bintrayUpload.
-            Task gitPush = project.getTasks().getByName(GitPlugin.GIT_PUSH_TASK);
-            gitPush.mustRunAfter(bintrayUpload.getTaskDependencies());
+        project.allprojects(subproject -> {
+            subproject.getPlugins().withType(ShipkitBintrayPlugin.class, plugin -> {
+                Task bintrayUpload = subproject.getTasks().getByName(ShipkitBintrayPlugin.BINTRAY_UPLOAD_TASK);
+                performRelease.dependsOn(bintrayUpload);
 
-            //bintray upload after git push so that when git push fails we don't publish jars to bintray
-            //git push is easier to undo than deleting published jars (not possible with Central)
-            bintrayUpload.mustRunAfter(gitPush);
+                //bintray upload after git push so that when git push fails we don't publish jars to bintray
+                //git push is easier to undo than deleting published jars (not possible with Central)
+                bintrayUpload.mustRunAfter(gitPush);
 
-            final BintrayExtension bintray = subproject.getExtensions().getByType(BintrayExtension.class);
-            deferredConfiguration(subproject, () -> {
-                UpdateReleaseNotesTask updateNotes = (UpdateReleaseNotesTask) project.getTasks().getByName(ReleaseNotesPlugin.UPDATE_NOTES_TASK);
-                String userSpecifiedRepo = conf.getLenient().getReleaseNotes().getPublicationRepository();
-                if (userSpecifiedRepo != null) {
-                    updateNotes.setPublicationRepository(userSpecifiedRepo);
-                } else {
-                    updateNotes.setPublicationRepository(BintrayUtil.getRepoLink(bintray));
-                }
+                final BintrayExtension bintray = subproject.getExtensions().getByType(BintrayExtension.class);
+                deferredConfiguration(subproject, () -> {
+                    UpdateReleaseNotesTask updateNotes = (UpdateReleaseNotesTask) project.getTasks().getByName(ReleaseNotesPlugin.UPDATE_NOTES_TASK);
+                    String userSpecifiedRepo = conf.getLenient().getReleaseNotes().getPublicationRepository();
+                    if (userSpecifiedRepo != null) {
+                        updateNotes.setPublicationRepository(userSpecifiedRepo);
+                    } else {
+                        updateNotes.setPublicationRepository(BintrayUtil.getRepoLink(bintray));
+                    }
+                });
             });
-        }));
+
+            subproject.getPlugins().withType(JavaBintrayPlugin.class, plugin -> {
+                //Making git push run as late as possible because it is an operation that is hard to reverse.
+                //Git push will be executed after all tasks needed by bintrayUpload
+                // but before bintrayUpload.
+                //Using task path as String because the task comes from maven-publish new configuration model
+                // and we cannot refer to it in a normal way, by task instance.
+                String mavenLocalTask = subproject.getPath() + ":" + MAVEN_LOCAL_TASK;
+                gitPush.mustRunAfter(mavenLocalTask);
+            });
+        });
     }
 }

--- a/subprojects/testDownstream/testDownstream.gradle
+++ b/subprojects/testDownstream/testDownstream.gradle
@@ -3,4 +3,4 @@ apply plugin: 'base' // for clean task, TODO ms: push it to the plugin, make 'ch
 apply plugin: 'org.shipkit.test-downstream'
 
 testDownstream.addRepository("https://github.com/mockito/shipkit-example")
-testDownstream.addRepository("https://github.com/mockito/mockito")
+//testDownstream.addRepository("https://github.com/mockito/mockito")

